pool/app/experiment/entity.ml

```diff
@@ -190,6 +190,37 @@ module Public = struct
     ; smtp_auth_id : Email.SmtpAuth.Id.t option
     }
   [@@deriving eq, show]
+
+  let create
+    ?description
+    ?language
+    ?experiment_type
+    ?smtp_auth_id
+    id
+    public_title
+    direct_registration_disabled
+    =
+    { id
+    ; public_title
+    ; description
+    ; language
+    ; direct_registration_disabled
+    ; experiment_type
+    ; smtp_auth_id
+    }
+  ;;
+
+  let update_direct_registration_disabled (m : t) direct_registration_disabled =
+    { m with direct_registration_disabled }
+  ;;
+
+  let id (m : t) = m.id
```

Comment:
Here, it's a bit different because the fields of the main experiment type are already derived. An error is shown for the public ones:

```
Error (warning 41): id belongs to several types: t/1 t/2
The first one was selected. Please disambiguate if this is wrong.
```